etc/cmake/options.cmake

```diff
@@ -241,3 +241,8 @@ endif()
 if(OT_PLATFORM_UDP AND OT_UDP_FORWARD)
     message(FATAL_ERROR "OT_PLATFORM_UDP and OT_UDP_FORWARD are exclusive")
 endif()
+
+option(OT_REFERENCE_DEVICE "enable reference device")
```

Comment:
@edmont We tend to avoid local changes to OpenThread core sources. Please open a PR towards upstream OpenThread with changes in this file.